python/ray/data/_internal/datasource/iceberg_datasource.py

```diff
@@ -190,6 +202,26 @@ def _get_read_task(
                 "number of files"
             )
 
+        # Get required properties for reading tasks - table IO, table metadata,
+        # row filter, case sensitivity,limit and projected schema. pre-apply
+        # them to `_get_read_task` through partial to avoid `self` reference
+        # which causes perfromance degradation during serialization
```

Comment:
```suggestion
        # Get required properties for reading tasks - table IO, table metadata,
        # row filter, case sensitivity,limit and projected schema to pass
        # them directly to `_get_read_task` to avoid capture of `self` reference
        # within the closure carrying substantial overhead invoking these tasks
        #
        # See XXX for more context
```

Reply:
@jimmyxie-figma can you please also file a ticket outlining details of this issue (that you already capture in the description) and link it here for future code reader reference

Reply:
@alexeykudinkin added a ticket to the comment